Frameworks/UIKit/UIApplicationMain.mm

```diff
@@ -384,8 +394,27 @@ void _UIApplicationShutdown() {
 }
 
 extern "C" void UIApplicationMainHandleProtocolEvent(IInspectable* protocolUri, const wchar_t* sourceApplication) {
-    WFUri* protocolResult = [WFUri createWith:protocolUri];
-    NSString* source = _bundleIdFromPackageFamilyName(sourceApplication);
+    id result = _createProjectionObject("WFUri", protocolUri, "protocol activation");
+
+    if (result) {
+        NSString* source = _bundleIdFromPackageFamilyName(sourceApplication);
+
+        [[UIApplication sharedApplication] _sendProtocolReceivedEvent:result source:source];
+    }
+}
+
+id _createProjectionObject(const char* className, IInspectable* source, const char* description) {
```

Comment:
Seems like a good idea.

---
In reply to: [112307470](https://github.com/Microsoft/WinObjC/pull/2569#discussion_r112307470) [](ancestors = 112307470)